grpc/src/main/java/io/stargate/grpc/service/SchemaHandler.java

```diff
@@ -42,25 +46,31 @@ public static void describeKeyspace(
         persistence.decorateKeyspaceName(query.getKeyspaceName(), GrpcService.HEADERS_KEY.get());
 
     try {
-      // TODO: check if null or access allowed?
       Keyspace keyspace = persistence.schema().keyspace(decoratedKeyspace);
+      if (keyspace == null) {
+        throw Status.NOT_FOUND.withDescription("Keyspace not found").asException();
+      }
 
       CqlKeyspaceDescribe.Builder describeResultBuilder = CqlKeyspaceDescribe.newBuilder();
       CqlKeyspace.Builder cqlKeyspaceBuilder = CqlKeyspace.newBuilder();
       cqlKeyspaceBuilder.setName(keyspace.name());
 
-      // TODO: Persistence implementation doesn't set replication strategy
-      //    Map<String, String> replication = new HashMap<String, String>(keyspace.replication());
-      //    if (replication.containsKey("class")) {
-      //      cqlKeyspaceBuilder.setReplicationStrategy(replication.remove("class"));
-      //      for (String datacenter : replication.keySet()) {
-      //        cqlKeyspaceBuilder.addReplicationFactors(
-      //            CqlDatacenterReplication.newBuilder()
-      //                .setDatacenterName(datacenter)
-      //                .setReplicationFactor(Integer.parseInt(replication.get(datacenter)))
-      //                .build());
-      //      }
-      //    }
+      HashMap<String, String> replication = new HashMap<String, String>(keyspace.replication());
+      if (replication.containsKey("class")) {
+        String strategyName = replication.remove("class");
+        if (strategyName.equals("SimpleStrategy")) {
+          cqlKeyspaceBuilder.putOptions("replication", Replication.simpleStrategy(1).toString());
```

Comment:
I know it defaults to `1` but could it ever be anything else?

Reply:
It supports `N` but in practice it's usually `1`.
https://docs.datastax.com/en/cassandra-oss/3.0/cassandra/architecture/archDataDistributeReplication.html#archDataDistributeReplication__simpleStrategy